Bottom: d6c06a3107d717d0929290faa73d391f3b7db210
Top:    1f0c479d29f5ee706d6ef97d4d2eda70e849ac0f
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-05 16:23:31 +0000

polyon1.c: Rework InsCntr to re-use poly_Create() rather than open-coding

This allows us to ensure common initialisation is performed in poly_Create().

This commit removes the single case where the memory allocation macro MemGet
was called, which will be removed in a later commit.


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 1498462..fff6a7a 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1203,23 +1203,25 @@ InsCntr (jmp_buf * e, PLINE * c, POLYAREA ** dst)
 {
   POLYAREA *newp;
 
+  newp = poly_Create ();
+  if (newp == NULL)
+    error(err_no_memory);
+
   if (*dst == NULL)
     {
-      MemGet (*dst, POLYAREA);
-      (*dst)->f = (*dst)->b = *dst;
-      newp = *dst;
+      newp->f = newp->b = newp;
+      *dst = newp;
     }
   else
     {
-      MemGet (newp, POLYAREA);
       newp->f = *dst;
       newp->b = (*dst)->b;
       newp->f->b = newp->b->f = newp;
     }
+
+  c->next = NULL;
   newp->contours = c;
-  newp->contour_tree = r_create_tree (NULL, 0, 0);
   r_insert_entry (newp->contour_tree, (BoxType *) c, 0);
-  c->next = NULL;
 }				/* InsCntr */
 
 static void
